boot/bootutil/src/bootutil_misc.c

```diff
@@ -139,6 +147,12 @@ boot_status_off(const struct flash_area *fap)
 
     off_from_end = boot_trailer_sz(elem_sz);
 
+#if MCUBOOT_SWAP_USING_SCRATCH
```

Comment:
See my dislike for this how this get fixed, rather another call boot_trailer_xyz_sz() should be provided instead

Reply:
As far as I could see the only place that needs to know the size of the trailer inside scratch was here, so another call would just be overkill that should not be used anywhere else. Do you have any plans on using it?

Reply:
I don't have plan to reuse new call. It is just about legibility of the code - how the trailer size is calculated.
Contribution to the size from the status is much smaller in case of scratch are, and this is what is subtracted here.

Don't treat my comments as a no-go.

Reply:
It's a fair assessment, I will do it as suggested later today once I'm done with work for the day. I was also curious to understand a bit better why this did not impact the swap with scratch operation the way it was. Sorry that this was a bit rushed.

Reply:
#1246 should address your concerns. I took a look at the effect of the bug and the swap operation was indeed writing at the wrong offset inside scratch, but the right amount of data, which is possibly still caught by the calculations that are done before starting a swap, but anyway it's still weird we didn't catch that before...